src/coreclr/System.Private.CoreLib/src/System/Variant.cs

```diff
@@ -432,5 +545,28 @@ internal static void MarshalHelperCastVariant(object pValue, int vt, ref Variant
                 };
             }
         }
+
+        internal static class SystemColorTranslatorAccess
+        {
+            private static readonly MethodInfo s_toOleMethod;
+            private static readonly MethodInfo s_fromOleMethod;
+
+#pragma warning disable CA1810 // Initialize reference type static fields inline
+            static SystemColorTranslatorAccess()
+#pragma warning restore CA1810
+            {
+                // Reflection pattern recognized by trimmer
+                Type? colorTranslatorType = Type.GetType("System.Drawing.ColorTranslator, System.Drawing");
+                Debug.Assert(colorTranslatorType != null);
+                s_toOleMethod = colorTranslatorType.GetMethod("ToOle", BindingFlags.Public | BindingFlags.Static)!;
+                s_fromOleMethod = colorTranslatorType.GetMethod("FromOle", BindingFlags.Public | BindingFlags.Static)!;
+            }
+
+            public static uint ConvertSystemColorToOleColor(object colorObj)
+                => (uint)(int)s_toOleMethod.Invoke(null, [colorObj])!;
+
+            public static object ConvertOleColorToSystemColor(uint oleColor)
+                => s_fromOleMethod.Invoke(null, [(int)oleColor])!;
```

Comment:
Sanity check here:

The unmanaged implementation only matches `System.Drawing.Color` by name. It can allow System.Drawing.Primitives loaded in different ALC. The ColorTranslator used should always be in default ALC.

However, a malformed structure also named `System.Drawing.Color` may be manipulated as color and cause undefined effect.

Supporting multiple ALCs is reasonable. How should it be dealed here? Using function pointers may require defining a struct with same layout as Color since it's used at return value.

Reply:
This would need tests to validate that the behavior. It is not obvious to me how ALCs are handled by the current implementation.

In any case, I do not think we want to have multiple paths that load System.Drawing.Color for interop. If you are changing it, all places need to use the new path.

Reply:
Sure. I've almost done the next PR that cleans up Variant totally, but it's still unlikely to include the (only) other case invoking System.Drawing.Color.

It's used by IDispatch marshaler. Is it CLR to COM *early* bind worker?